### PR TITLE
feat: Multi region support for accelerator

### LINKED ISSUE #675: Feature-Request: Multi-Region -> Hub-and-spoke topology
### Describe the feature end to end, including deployment scenario details under which the feature would occur.

Assuming I want to deploy the Hub-and-spoke network topology and I want to deploy multiple regions, each with its own Hub network.

The recommendation here: https://github.com/Azure/ALZ-Bicep/issues/193#issuecomment-1075620583 is to simply deploy the `hubNetworking` module in each region.

Of course, I could do that. But then, the Hub networks are not automatically peered. And by now, Multi-Region is not supported by the [_Accelerator_](https://github.com/Azure/ALZ-Bicep/wiki/Accelerator).

My suggestion is:
- add an additional parameter to the `hubNetwork` module, where one can add other Hub networks to peer to
- modify the _Accelerator_ pipeline to watch `config/custom-parameters/hubNetworking.parameters.all.*.json` (something with a wildcard, to be able to add the region to the filename)
- modify the _Accelerator_ Powershell script `Deploy-ALZHub-HubAndSpoke.ps1` to loop over all `hubNetworking.parameters` files

### Why is this feature important. Describe why this would be important for your organization and others. Would this impact similar orgs in the same way?

If I use IAC to deploy the CAF, I don't want any manual steps in the deployment. I'd like to be able to configure the peering between the Hubs in the parameter file too.

### Please provide the correlation id associated with your error or bug.

xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

### Can you describe any alternatives that you have taken since this feature does not exist?

_No response_

### Feature Implementation

_No response_

### Check previous GitHub issues

- [X] I have searched the issues for this item and found no duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sebassem**
# Overview/Summary

Creates necessary changes to ALZ-PowerShell-Config for Accelerator v2 to support multi-region deployments. Also, includes relevant parameter file for VWAN and hubnetworking module that supports deployments to 2 regions.

Finally, adds notice for the upcoming change to zone redundant configurations by default.

## Related Issues/Work Items

Closes #675
Fixes [AB#34223](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/34223)

## This PR fixes/adds/changes/removes

- Added an additional hubNetworking module for 2-region deployments
- Added an additional parameters file in case multi-region is selected, to deploy both regions
- Updated some parameters' values to include the region to make a clear distinction between them like bastion and nsgs
- Implement vnet peering between the 2 hubs
- Add notice for zone redundant configuration date in Q4 2024.
- 
### Breaking Changes

None

## Testing Evidence

Replace this with any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [x] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: accelerator/.config/ALZ-Powershell-Auto.config.json [hunk 1]
```diff
@@ -284,6 +298,35 @@
...
+              "Destination": "Parameters"
+            },
+            {
+              "Name": "parPolicyAssignmentParameters.value.ascExportResourceGroupLocation.value",
+              "Destination": "Parameters"
+            },
+            {
+              "Name": "parVirtualWanHubs.value[0].parHubLocation",
+              "Destination": "Parameters"
+            },
```
Comment (jaredfholgate) [inline accelerator/.config/ALZ-Powershell-Auto.config.json:322]: I think these ones need to be removed or updated?

File: accelerator/.config/ALZ-Powershell-Auto.config.json [hunk 2]
```diff
@@ -560,6 +636,105 @@
             }
           ]
         },
+        "parAzBastionName": {
```
Comment (jaredfholgate) [inline accelerator/.config/ALZ-Powershell-Auto.config.json:639]: I understand the reason for adding these. I'm just wondering if we are potentially impacting existing users upgrading a single region deployment?

File: accelerator/.config/ALZ-Powershell-Auto.config.json (hunk 1 above)
Comment (oZakari) [reply accelerator/.config/ALZ-Powershell-Auto.config.json:322]: Fixed

**Review by oZakari — APPROVED** [review]

**Pull request merged**